crates/libtiny_tui/src/key_map.rs

```diff
@@ -232,30 +233,36 @@ impl<'de> Deserialize<'de> for MappedKey {
                         "pgdown" => Key::PageDown,
                         "pgup" => Key::PageUp,
                         "tab" => Key::Tab,
-                        "up" => Key::Arrow(Arrow::Up),
-                        "down" => Key::Arrow(Arrow::Down),
-                        "left" => Key::Arrow(Arrow::Left),
-                        "right" => Key::Arrow(Arrow::Right),
+                        arrow if Arrow::from_str(arrow).is_ok() => {
+                            Key::Arrow(Arrow::from_str(arrow).unwrap())
+                        }
+                        f if FKey::from_str(f).is_ok() => Key::FKey(FKey::from_str(f).unwrap()),
                         ch => Key::Char(single_key(ch)?),
                     },
                     Some((k1, k2)) => match k1 {
                         "alt" => match k2 {
-                            "up" => Key::AltArrow(Arrow::Up),
-                            "down" => Key::AltArrow(Arrow::Down),
-                            "left" => Key::AltArrow(Arrow::Left),
-                            "right" => Key::AltArrow(Arrow::Right),
+                            arrow if Arrow::from_str(arrow).is_ok() => {
+                                Key::AltArrow(Arrow::from_str(arrow).unwrap())
+                            }
+                            f if FKey::from_str(f).is_ok() => Key::AltF(FKey::from_str(f).unwrap()),
```

Comment:
@osa1 can you think of a cleaner way to do these mappings?

Reply:
@trevarj We should not have a `FromStr` in `term_input` as parsing is specific to one downstream. For example, you could imagine another user that wants to map `"Up"` instead of `"up"` to `Arrow::Up`, or parse case sensitively. As far as I'm aware currently there's one user for term_input which is tiny, but it's still a good idea to keep tiny specific code out of term_input.

Regarding `is_ok()` and `unwrap()` parts: I think something like this would be fine:

```rust
match v {
    "backspace" => Key::Backspace,
    "del" => Key::Del,
    "end" => Key::End,
    "esc" => Key::Esc,
    "home" => Key::Home,
    "pgdown" => Key::PageDown,
    "pgup" => Key::PageUp,
    "tab" => Key::Tab,
    other => {
        if let Some(arrow) = parse_arrow(other) {
            ...
        } else if let Some(f_key) = parse_f_key(other) {
            ...
        } else {
            ...
        }
    }
}
```

Reply:
@osa1 that's perfect, thanks. Made the change.